src/dynamic_contract.rs

```diff
@@ -0,0 +1,194 @@
+extern crate bincode;
+extern crate generic_array;
+
+use libloading::{Library, Symbol};
+use std::path::PathBuf;
+use std::result;
+use system_contract::AccountInfo;
+
+#[cfg(debug_assertions)]
+const CARGO_PROFILE: &str = "debug";
+
+#[cfg(not(debug_assertions))]
+const CARGO_PROFILE: &str = "release";
+
+/// Dynamic link library prefix
+#[cfg(unix)]
+const PLATFORM_FILE_PREFIX: &str = "lib";
+/// Dynamic link library prefix
+#[cfg(windows)]
+const PLATFORM_FILE_PREFIX: &str = "";
+/// Dynamic link library file extension specific to the platform
+#[cfg(any(target_os = "macos", target_os = "ios"))]
+const PLATFORM_FILE_EXTENSION: &str = "dylib";
+/// Dynamic link library file extension specific to the platform
+#[cfg(all(unix, not(any(target_os = "macos", target_os = "ios"))))]
+const PLATFORM_FILE_EXTENSION: &str = "so";
+/// Dynamic link library file extension specific to the platform
+#[cfg(windows)]
+const PLATFORM_FILE_EXTENSION: &str = "dll";
+
+/// Creates a platform-specific file path
+fn create_library_path(name: &str) -> PathBuf {
+    let mut path = PathBuf::new();
+    path.push("target");
+    path.push(CARGO_PROFILE);
+    path.push("deps");
+    path.push(PLATFORM_FILE_PREFIX.to_string() + name);
+    path.set_extension(PLATFORM_FILE_EXTENSION);
+    path
+}
+
+// All contracts export a symbol named process()
+const ENTRYPOINT: &str = "process";
+type Entrypoint = unsafe extern "C" fn(infos: &mut Vec<AccountInfo>, data: &[u8]);
+
+#[derive(Debug)]
+pub enum DynamicContract {
+    /// Native contract
+    /// * Transaction::keys[0..] - contract dependent
+    /// * name - name of the contract, translated to a file path of the contract module
+    /// * userdata - contract specific user data
+    Native { name: String, library: Library },
+    /// BPF contract
+    /// * Transaction::keys[0..] - contract dependent
+    /// * TODO BPF specific stuff
+    /// * userdata - contract specific user data
+    BPF { userdata: Vec<u8> },
```

Comment:
Rust convention is to only capitalize the first letter of acronyms. Can you make this `Bpf`?